### PR TITLE
chore(imports): Reorganize boilerplate imports and adhere to `import/order` linter rule

### PR DESCRIPTION
**Pull request by markrickert**
## Description

I noticed the ignite imports growing inconsistent over time sometimes using the typescript `@/*` import path and other times using relative paths. This PR is an attempt to solve that problem by implementing and enforcing the [`import/order` eslint rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md).

This is a fairly large reorganization of the imports section of the boilerplate using the `import/order` eslint rule to separate and organize the import sections of the boilerplate codebase. 

1. I moved the `prettier`, `jest`, and `eslint` configuration for the main ignite-cli out into their own files from insied the main `package.json` file. No changes to the actual configs
2. Added to group and sort import statements automatically for easier groking of the codebase. 
3. We had a typescript "assets" alias that wasn't being used anywhere. Changed all relative paths to `*../assets/*` to use the typescript alias `assets/*`
4. Searched the boilerplate codebase for all relative paths going *up* one or more directories and converted them to use the `@/*` typescript alias. for instance, `import { $styles } from "../theme"` became `import { $styles } from "@/theme"`. This was NOT a find/replace-all, but a careful and methodical replacement throughout the boilerplate, leaving some spots with the relative import where it made sense (mostly in the `app.tsx` and in the demo screen files referencing each other).
5. Updated documentation to be consistent with the imports for both assets and relative paths.

## Checklist

- [x] `README.md` and other relevant documentation has been updated with my changes
- [x] I have manually tested this, including by generating a new app locally ([see docs](https://docs.infinite.red/ignite-cli/contributing/Contributing-To-Ignite/#testing-changes-from-your-local-copy-of-ignite)).

### CODE REVIEW

File: boilerplate/app/components/index.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export * from "./AutoImage"
 export * from "./Button"
 export * from "./Card"
+export * from "./EmptyState"
```
[the remark this answers is not in the document]
Comment (frankcalise) [reply boilerplate/app/components/index.ts:4]: were ignored ;D https://github.com/infinitered/ignite/pull/2956/files#diff-1271a3bdea0a935e34e0c9cfda248039016cc9bd1e3090ccf0320c63f4a36994L9

[the remark this answers is not in the document]
Comment (joshuayoes) [reply boilerplate/app/components/index.ts:4]: +1 to import aliases reducing dependency cycles

[the remark this answers is not in the document]
Comment (frankcalise) [reply boilerplate/app/components/index.ts:4]: Yeah wherever the components are imported they'd end up on separate lines and the `index.ts` in `app/components` gets deleted. But it also exists in other directories:

There's also one in the `models`, `navigators` and `screens` dir. Some `export *`s in theme.

And then we'd have to update the `ignite/templates` to not import from the barrel index as well.

Does that sound right, @coolsoftwaretyler ?

[the remark this answers is not in the document]
Comment (markrickert) [reply boilerplate/app/components/index.ts:4]: @joshuayoes I used `madge` to find the circular deps. I think adding a circular-deps test to the CI to make sure there are none is a good idea - #2959
